modules/__init__.py

```diff
@@ -1,38 +1,28 @@
 from modules.base import Module, noparts, check_solve_cmd, gif_append, gif_output
+import glob
+from os.path import join as pjoin
+from os.path import dirname as pdir
+from os.path import basename as pbase
+import importlib
 
-from modules.wires import Wires, ComplicatedWires, WireSequence
-from modules.button import Button
-from modules.keypad import Keypad
-from modules.simon import SimonSays
-from modules.whos_on_first import WhosOnFirst
-from modules.memory import Memory
-from modules.morse import MorseCode
-from modules.maze import Maze
-from modules.password import Password
+VANILLA_MODULE_LIST="wires/button/keypad/simonSays/whosOnFirst/memory/morseCode/complicatedWires/wireSequence/maze/password".split("/")
 
-from modules.connection_check import ConnectionCheck
-from modules.hexamaze import Hexamaze
-from modules.thirdbase import ThirdBase
+VANILLA_MODULES={}
+MODDED_MODULES={}
 
-VANILLA_MODULES = {
-    "wires": Wires,
-    "button": Button,
-    "keypad": Keypad,
-    "simonSays": SimonSays,
-    "whosOnFirst": WhosOnFirst,
-    "memory": Memory,
-    "morseCode": MorseCode,
-    "complicatedWires": ComplicatedWires,
-    "wireSequence": WireSequence,
-    "maze": Maze,
-    "password": Password,
-}
+for i in glob.glob(pjoin(pdir(__file__),"*.py")):
+    if pbase(i) in ["__init__.py","base.py"]:
+        continue
+    moduleName=pbase(i)[:-3]
+    moduleImport=importlib.import_module('modules.'+moduleName)
+    classObject=moduleImport.__module_class__
```

Comment:
I wonder if you could use a metaclass to make all subclasses of `Module` register automatically. That would make `__module_class__` unnecessary. It's definitely not a blocker for merging, though.

Reply:
Yeah i am just working on that :D